core/src/main/scala/cats/instances/list.scala

```diff
@@ -179,6 +179,19 @@ trait ListInstances extends cats.kernel.instances.ListInstances {
 
       override def collectFirstSome[A, B](fa: List[A])(f: A => Option[B]): Option[B] =
         fa.collectFirst(Function.unlift(f))
+
+      override val unit: List[Unit] = () :: Nil
```

Comment:
Given that this is being done with an eye towards optimization, my recommendation would be to follow the pattern used in Cats Effect to avoid the extra volatile read (on the object bitmap):

```suggestion
      override def unit: List[Unit] = _unit
      private[this] val _unit: List[Unit] = () :: Nil
```